LoopKitUI/Views/BasalScheduleEntryTableViewCell.swift

```diff
@@ -0,0 +1,242 @@
+//
+//  BasalScheduleEntryTableViewCell.swift
+//  LoopKitUI
+//
+//  Created by Pete Schwamb on 2/23/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+protocol BasalScheduleEntryTableViewCellDelegate: class {
+    func basalScheduleEntryTableViewCellDidUpdate(_ cell: BasalScheduleEntryTableViewCell)
+    func validateBasalScheduleEntryTableViewCell(_ cell: BasalScheduleEntryTableViewCell) -> Bool
+}
+
+private enum Component: Int, CaseIterable {
+    case time = 0
+    case value
+}
+
+class BasalScheduleEntryTableViewCell: UITableViewCell {
+
+    @IBOutlet weak var picker: UIPickerView!
+
+    @IBOutlet weak var pickerHeightConstraint: NSLayoutConstraint!
+
+    private var pickerExpandedHeight: CGFloat = 0
+
+    @IBOutlet weak var dateLabel: UILabel!
+
+    @IBOutlet weak var unitLabel: UILabel!
+
+    @IBOutlet weak var valueLabel: UILabel!
```

Comment:
All three labels can be made private; looks like their text is only updated via other functions or in response to property changes.